target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-eap1300.dts

```diff
@@ -174,6 +174,13 @@
 				label = "0:APPSBL";
 				reg = <0x000f0000 0x00090000>;
 				read-only;
+				compatible = "nvmem-cells";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				macaddr_orgdata_71f28: macaddr@71f28 {
```

Comment:
Are you sure about this?

Because, it makes no sense that MAC-s are stored in the U-Boot partition

Reply:
It is a bit weird.  I searched through all the memory and it's the only place I found with the MAC address, though.  I'll look again.  I realized that I hadn't verified the MAC addresses of the WIFI devices, so I need to check that anyway.

Reply:
You are right.  I booted the factory firmware, and the MACs are not what the 22.03.5 version was showing.  There is a MAC address reported by the u-boot, which is what I was using, but that is apparently fixed in the firmware and wrong.

The 22.03.5 version comes up with a MAC address on the ethernet devices that is whatever is left over in the device, which appears to be fixed.  The wifi macs come from 6 bytes into the configuration data, which appears to be some kind of default, but again is wrong.

The only place I could find the MAC address for eth0 is the u-boot-env partition, there is a macaddr= field there that is correct.  The wifi MAC addresses are the macaddr value incremented by 1 and 2, respectively.  There is a wanaddr= field that is what the eth device comes up as if you don't set anything, so I'm assuming u-boot is setting that.

u-boot is not setting the MAC in the device tree like it normally would.  I'm guessing in the OEM load some software comes up later, read the MAC address from the u-boot environment, sets the eth device, and generates the others for the wifi devices.  What can I do here?

Reply:
Check other dts, uboot env entry can be parsed as nvmem cell and used to reference mac addr... then with the help of nvmem-layout you can increment it by 1 and 2 for wifi. It done my times in other dts so you should be able to find many examples

keyword are nvmem-layout and uboot or uboot,env if i'm not wrong.

Reply:
https://github.com/openwrt/openwrt/blob/main/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-wax218.dts#L174

One example, but do note that this only works if the U-Boot variable name is ethaddr or ethNaddr for multiple addresses